examples_tests/common/CommonAPI.h

```diff
@@ -0,0 +1,287 @@
+		sp.flags = video::IGPURenderpass::ESDF_NONE;
+		sp.inputAttachmentCount = 0u;
+		sp.inputAttachments = nullptr;
+		sp.preserveAttachmentCount = 0u;
+		sp.preserveAttachments = nullptr;
+		sp.resolveAttachments = nullptr;
+
+		video::IGPURenderpass::SCreationParams rp_params;
+		rp_params.attachmentCount = 1u;
+		rp_params.attachments = &a;
+		rp_params.dependencies = nullptr;
+		rp_params.dependencyCount = 0u;
+		rp_params.subpasses = &sp;
+		rp_params.subpassCount = 1u;
+
+		return device->createGPURenderpass(rp_params);
+	}
+
+	template<size_t imageCount, size_t width, size_t height>
+	static auto createFBOWithSwapchainImages(const nbl::core::smart_refctd_ptr<nbl::video::ILogicalDevice>& device,
+		nbl::core::smart_refctd_ptr<nbl::video::ISwapchain> swapchain,
+		nbl::core::smart_refctd_ptr<nbl::video::IGPURenderpass> renderpass)->std::array<nbl::core::smart_refctd_ptr<nbl::video::IGPUFramebuffer>, imageCount>
+	{
+		using namespace nbl;
+		std::array<nbl::core::smart_refctd_ptr<video::IGPUFramebuffer>, imageCount> fbo;
+		auto sc_images = swapchain->getImages();
+		assert(sc_images.size() == imageCount);
+		for (uint32_t i = 0u; i < imageCount; ++i)
+		{
+			auto img = sc_images.begin()[i];
+			core::smart_refctd_ptr<video::IGPUImageView> view;
+			{
+				video::IGPUImageView::SCreationParams view_params;
+				view_params.format = img->getCreationParameters().format;
+				view_params.viewType = asset::IImageView<video::IGPUImage>::ET_2D;
+				view_params.subresourceRange.baseMipLevel = 0u;
+				view_params.subresourceRange.levelCount = 1u;
+				view_params.subresourceRange.baseArrayLayer = 0u;
+				view_params.subresourceRange.layerCount = 1u;
+				view_params.image = std::move(img);
+
+				view = device->createGPUImageView(std::move(view_params));
+				assert(view);
+			}
+
+			video::IGPUFramebuffer::SCreationParams fb_params;
+			fb_params.width = width;
+			fb_params.height = height;
+			fb_params.layers = 1u;
+			fb_params.renderpass = renderpass;
+			fb_params.flags = static_cast<video::IGPUFramebuffer::E_CREATE_FLAGS>(0);
+			fb_params.attachmentCount = 1u;
+			fb_params.attachments = &view;
+
+			fbo[i] = device->createGPUFramebuffer(std::move(fb_params));
+			assert(fbo[i]);
+		}
+		return fbo;
+	}
+
+	static void Submit(nbl::video::ILogicalDevice* device,
+		nbl::video::ISwapchain* sc,
+		nbl::core::smart_refctd_ptr<nbl::video::IGPUCommandBuffer>* cmdbuf,
+		nbl::video::IGPUQueue* queue,
+		nbl::video::IGPUSemaphore* imgAcqSemaphore,
+		nbl::video::IGPUSemaphore* renderFinishedSemaphore,
+		size_t imageCount,
+		uint32_t imgNum)
+	{
+		using namespace nbl;
+		video::IGPUQueue::SSubmitInfo submit;
+		{
+			auto* cb = cmdbuf[imgNum].get();
+			submit.commandBufferCount = 1u;
+			submit.commandBuffers = &cb;
+			video::IGPUSemaphore* signalsem = renderFinishedSemaphore;
+			submit.signalSemaphoreCount = 1u;
+			submit.pSignalSemaphores = &signalsem;
+			video::IGPUSemaphore* waitsem = imgAcqSemaphore;
```

Comment:
briliant, you round robin commandbuffers so you dont overstep... but you dont round robin the semaphores XD